### PR TITLE
Feature/use hcaptcha if bot

### PR DESCRIPTION
**Pull request by MGibson1**
# Overview

Add captcha challenges to login and registration pages.

# Files Changed

* **login.component**: add id to content component for resizing on captcha required. Add captcha div
* **register.component**: Add captcha div and initialize `captchaProtectedComponent` ancestor.
* **misc.scss**: style captcha div
* **pages.scss**: smooth width transition of login component when captcha is revealed.

# Screen shots

## pre-captcha
image

## captcha
image

## captcha with images
image

## captcha complete
image

## Register pre-captcha
image

## Register with catpcha
image

## Register with images
image

## Register complete
image

### CODE REVIEW

File: src/app/accounts/login.component.html [hunk 1]
```diff
@@ -22,6 +22,9 @@
                         </a>
                     </div>
                 </div>
+                <div class="box-content-row" [hidden]="!showCaptcha()">
+                    <iframe id="hcaptcha_iframe" height=80></iframe>
```
Comment (cscharf) [inline src/app/accounts/login.component.html:26]: attribute double quotes please.

File: src/app/accounts/register.component.html [hunk 2]
```diff
@@ -61,6 +61,9 @@ <h1>{{'createAccount' | i18n}}</h1>
                     <label for="hint">{{'masterPassHint' | i18n}}</label>
                     <input id="hint" type="text" name="Hint" [(ngModel)]="hint">
                 </div>
+                <div class="box-content-row" [hidden]="!showCaptcha()">
+                    <iframe id="hcaptcha_iframe" height=80></iframe>
```
Comment (cscharf) [inline src/app/accounts/register.component.html:65]: same

**Review by cscharf — COMMENTED** [review]
Looks good, like the other PR, just the attributes need some quotes

**Review by cscharf — APPROVED** [review]

**Comment by MGibson1** [comment]
Sorry @cscharf, needed another jslib update :(

**Review by cscharf — APPROVED** [review]

**Pull request merged**